simpeg/directives/_regularization.py

```diff
@@ -403,17 +403,22 @@ def stopping_criteria(self):
                 )
             return True
 
-        # Check if the function has changed enough
-        f_change = np.abs(self.metrics.f_old - phim_new) / (self.metrics.f_old + 1e-12)
+        # Get how much the regularization has changed from previous value
+        f_change = (
+            np.abs(self.metrics.f_old - phim_new) / self.metrics.f_old
+            if self.metrics.f_old != 0.0
+            else np.inf
+        )
+
+        # Get target chi factor minus current chi factor
+        chi_factor_diff = 1.0 - self.invProb.phi_d / self.misfit_from_chi_factor(
+            self.chifact_target
+        )
```

Comment:
One other thing: shouldn't the hard coded 1.0 be the `self.chifact_target`? Or should it always be 1.0?
```suggestion
        chi_factor_diff = self.chifact_target - self.invProb.phi_d / self.misfit_from_chi_factor(
            self.chifact_target
        )
```

Reply:
I dont think so? The 1 in there is equivalent as doing 

`(phi_d* - phi_d) / phi_d*`

meaning the relative deviation from the target. The function `self.misfit_from_chi_factor` converts the chifact to phi_d value.

Reply:
You're right! My bad. I think what we can do is to rename the variable to make it reflect better what value it holds.


One minor thing: I get that by definition the chi factor is going to have the units of `phi_d / N`, but we always assume that the chi factor is unitless, meaning that `phi_d` is also unitless (because we are dividing by the uncertainties). This becomes quite clear when we look at the stopping criteria in `TargetMisfit` where we check if `phi_d < chifact * phi_d_star`, meaning that `chifact` is assumed to be always unitless. Nothing to take action on, but just wanted to mention it.

Reply:
Yep, you are right - that would make phi_d already unitless.